projects/graph/graph.py

```diff
@@ -1,48 +1,115 @@
 """
 Simple graph implementation
 """
+
 from util import Stack, Queue  # These may come in handy
 
+
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
+
     def __init__(self):
         self.vertices = {}
+
     def add_vertex(self, vertex):
         """
         Add a vertex to the graph.
         """
+        self.vertices[vertex] = set()
         pass  # TODO
+
     def add_edge(self, v1, v2):
         """
         Add a directed edge to the graph.
         """
+        if v1 not in self.vertices:
+            self.add_vertex(v1)
+        if v2 not in self.vertices:
+            self.add_vertex(v2)
+
+        self.vertices[v1].add(v2)
+        self.vertices[v2].add(v2)
```

Comment:
This line should be removed because the v2 vertex will be added to its own set.

Reply:
This was something Tim went over so maybe I need a better understanding as to why he did this.